cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/EventSubscriber.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.events;
+
+import com.google.common.util.concurrent.AbstractScheduledService;
+
+/**
+ * Abstract class for an event subscriber.
+ * Each EventSubscriber will receive and process incoming events.
+ */
+public abstract class EventSubscriber extends AbstractScheduledService {
+
+  /**
+   * Initialize this handler.
+   *
+   * @return true if successful
+   */
+  public abstract boolean initialize();
```

Comment:
For simplicity, I'm thinking of making this method void, and catch/log if it throws an exception in EventSubscriberManager. Otherwise, we can treat it as successfully initialized.

Reply:
What exception should it throw? If it can't load any event readers? In the StartProgramEventSubscriber, if one event reader can't load, it doesn't impact the other ones: https://github.com/cdapio/cdap/blob/e87b96137107f3f22283211f4c7446c21f1a45a7/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/StartProgramEventSubscriber.java#L83

Instead, should I just throw an exception and handle it in the manager?